extension/src/tdigest.rs

```diff
@@ -245,6 +247,89 @@ CREATE AGGREGATE timescale_analytics_experimental.tdigest(size int, value DOUBLE
 );
 "#);
 
+#[pg_extern(schema = "timescale_analytics_experimental")]
+pub fn tdigest_compound_trans(
+    state: Option<Internal<InternalTDigest>>,
+    value: Option<timescale_analytics_experimental::TDigest<'static>>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal<InternalTDigest>> {
+    unsafe {
+        in_aggregate_context(fcinfo, || {
+            match (state, value) {
+                (a, None) => a,
+                (None, Some(a)) => Some(a.to_internal_tdigest().into()),
+                (Some(a), Some(b)) => {
+                    Some(InternalTDigest::merge_digests(
+                            vec![a.deref().clone(), b.to_internal_tdigest()]  // TODO: TDigest merge with self
+                        ).into())
+                }
+            }
+        })
+    }
+}
+
+#[pg_extern(schema = "timescale_analytics_experimental")]
+pub fn tdigest_compound_combine(
+    state1: Option<Internal<InternalTDigest>>,
+    state2: Option<Internal<InternalTDigest>>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal<InternalTDigest>> {
+    unsafe {
+        in_aggregate_context(fcinfo, || {
+            match (state1, state2) {
+                (None, None) => None,
+                (None, Some(state2)) => Some(state2.clone().into()),
+                (Some(state1), None) => Some(state1.clone().into()),
+                (Some(state1), Some(state2)) => {
+                    Some(InternalTDigest::merge_digests(
+                            vec![state1.deref().clone(), state2.deref().clone()]  // TODO: TDigest merge with self
```

Comment:
Maybe I missed this, but what happens if you try to merge digests that have different sizes specified? Will there be an error somewhere?

Reply:
I had thought this was handled in the merge_digests code, but looking closer, such merges are actually explicitly allowed.  This actually raises the question of whether we should actually support mixed size merge?  I've added asserts everywhere to fail for now, but we can discuss this further.